frontend/src/store/course/actions.ts

```diff
@@ -232,7 +232,7 @@ const updateIsMentorChoosingEnabled = createAsyncThunk<
     course: { mentors, mentor },
   } = getState();
   const isMentor = mentors.some(
-    (mentor) => mentor.id === (user as UserWithPermissions).id,
+    (mentor) => mentor?.id === (user as UserWithPermissions)?.id,
```

Comment:
```suggestion
    (mentor) => mentor.id === (user as UserWithPermissions).id,
```
in this case, why are we casting a type here?